spec/systems/sessions_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Sessions', type: :system do
+  describe 'login' do
+    context 'validates screen' do
+      before(:each) do
+        visit login_path
+      end
+
+      it 'login screen should show correct field and button' do
+        within 'form' do
+          expect(page).to have_field('username')
+          expect(page).to have_field('password')
+          expect(page).to have_button(I18n.t('auth.login'))
+        end
+      end
+
+      it 'login screen should show correct label and components' do
+        expect(page).to have_selector('nav')
+        expect(page).to have_content(I18n.t('auth.login'))
+        expect(page).to have_content("#{I18n.t('app.do_not_have_account_yet')} #{I18n.t('auth.signup')}")
+
+        within 'form' do
+          expect(page).to have_content(I18n.t('auth.username'))
+          expect(page).to have_content(I18n.t('auth.password'))
+        end
+      end
+    end
+
+    context 'happy path' do
+      it 'given valid data should redirect to index page with greeting message' do
+        user = Fabricate(:user, username: 'nimblehq', password: 'password', password_confirmation: 'password')
+
+        visit login_path
+
+        within 'form' do
+          fill_in('username', with: user[:username])
+          fill_in('password', with: 'password')
+
+          click_button(I18n.t('auth.login'))
+        end
+
+        expect(current_path).to eql(root_path)
+
+        within 'nav' do
+          expect(page).to have_content("#{I18n.t('app.greeting')} #{user[:username]}")
+        end
+      end
+
+      it 'clicks at signup link should redirect to correct page' do
```

Comment:
I think we don't need to test this one because
- It's too small for a feature test
- We are testing the `click to a link` behavior 🤔

Reply:
For second bullet, do you mean `we are not testing the click to link behavior`?

Reply:
Yes @gutakk as click to the link the redirect the user to somewhere, that is the normal web behaviour

Reply:
Fixed